app/src/app/[locale]/serverAction/page.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
```

Comment:
Are there alternative approaches where the entire page doesn't need opted out of being a server component?

Reply:
No expert at this, but I believe you can have server only forms - I'm thinking of the way server actions support progressive enhancement and you can submit a form with JS turned off and the client not needing to be hydrated - but then you don't demo the client hooks like `useFormStatus` and `useFormState`. For a server form, instead of `useFormState` you would just call the server action function directly in the form's action attribute - but you can't input text I guess.

I'm not sure if another server (second) example makes sense for comparison? IE - Just call a server action from a server component page - and pass the resulting data down into a client component that displays it.

Just wanted to get things started on this cause I think it would have helped on our project if we had these examples.

Reply:
I think one alternative is to have a separate component for the part that depends on the client-only hook. I think for most forms it's likely just the submit button, so it might be something like:

```ts
// components/SubmitButton.tsx
"use client";

function SubmitButton(props) {
  const { pending } = useFormStatus();

  return <button disabled={pending} type="submit">{props.label}</button>;
}
```

I think that would then allow the page to be a server component with a small client-side island.

Reply:
@sawyerh this makes sense and we can try this - but we won't be able to demo `useFormState` without making the form (and text inputs) client components. I updated it this way if you want to take a peek .

It can work with the form in a server component (with some eslint rules turned off). Lmk if we want multiple examples or something.